easybuild/easyblocks/f/freesurfer.py

```diff
@@ -48,24 +48,46 @@ def extra_options():
 
     def install_step(self):
         """Custom installation procedure for FreeSurfer, which includes installed the license file '.license'."""
-        txt = super(EB_FreeSurfer, self).install_step()
-        try:
-            f = open(os.path.join(self.installdir, '.license'), 'w')
-            f.write(self.cfg['license_text'])
-            f.close()
-        except IOError, err:
-            raise EasyBuildError("Failed to install license file: %s", err)
+        super(EB_FreeSurfer, self).install_step()
+        write_file(os.path.join(self.installdir, '.license'), self.cfg['license_text'])
+
+    def make_module_guesses(self):
+        """Include correct subdirectories to $PATH for FreeSurfer."""
+        guesses = super(EB_FreeSurfer, self).make_module_guesses()
+
+        guesses['PATH'] = ['bin', 'fsfast/bin', 'mni/bin', 'tktools']
+
+        return guesses
 
     def make_module_extra(self):
-        """Add setting of FREESURFER_HOME in module."""
+        """Define FreeSurfer-specific environment variable in generated module file."""
         txt = super(EB_FreeSurfer, self).make_module_extra()
-        txt += self.module_generator.set_environment("FREESURFER_HOME", self.installdir)
+
+        freesurfer_vars = {
+            'FMRI_ANALYSIS_DIR': os.path.join(self.installdir, 'fsfast'),
+            'FS_OVERRIDE': '0',
+            'FSF_OUTPUT_FORMAT': 'nii.gz',
+            'FSFAST_HOME': os.path.join(self.installdir, 'fsfast'),
+            'FREESURFER_HOME': self.installdir,
+            'FUNCTIONALS_DIR': os.path.join(self.installdir, 'sessions'),
+            'MNI_DIR': os.path.join(self.installdir, 'mni'),
+            'MNI_DATAPATH': os.path.join(self.installdir, 'mni', 'data'),
+            'MINC_BIN_DIR': os.path.join(self.installdir, 'mni', 'bin'),
+            'MINC_LIB_DIR': os.path.join(self.installdir, 'mni', 'lib'),
+            'MNI_PERL5LIB': os.path.join(self.installdir, 'mni', 'lib', 'perl5', '5.8.5'),
+            'PERL5LIB': os.path.join(self.installdir, 'mni', 'lib', 'perl5', '5.8.5'),
+            'OS': 'linux',
+            'SUBJECTS_DIR': os.path.join(self.installdir, 'subjects'),
```

Comment:
Hmm, this looks like it may be a site-specific thing?

The problem with letting the easyblock set this is that it's not easy to prevent it from ending up in the generated module file, and the user would have to do an `unset FIX_VERTEX_AREA`.

Any upstream reference to this?

Reply:
@akesandgren Are you sure `FIX_VERTEX_AREA` is still relevant? This is from the release notes for `FreeSurfer` 3.0.1 (cfr. https://surfer.nmr.mgh.harvard.edu/fswiki/PreviousReleaseNotes):

```
FIX_VERTEX_AREA no longer need be set in the environment outside of recon-all. 
```

Reply:
Ok, but i wonder what they did to version numbering then... because my old (from 2015) tar file is called, freesurfer-Linux-centos6_x86_64-stable-pub-v5.3.0.tar.gz. Quite a lot higher then your current 3.0.1...

Reply:
Ohh, it might have been left there since some very much older version and i just copied the module file...
Then just ignore it.